bevy_renet/src/transport.rs

```diff
@@ -104,23 +104,51 @@ impl NetcodeClientPlugin {
     }
 }
 
-pub fn client_connected(transport: Option<Res<NetcodeClientTransport>>) -> bool {
-    match transport {
+pub fn client_connected() -> impl FnMut(Option<Res<NetcodeClientTransport>>) -> bool {
+    |transport| match transport {
         Some(transport) => transport.is_connected(),
         None => false,
     }
 }
 
-pub fn client_diconnected(transport: Option<Res<NetcodeClientTransport>>) -> bool {
-    match transport {
+pub fn client_diconnected() -> impl FnMut(Option<Res<NetcodeClientTransport>>) -> bool {
+    |transport| match transport {
         Some(transport) => transport.is_disconnected(),
         None => true,
     }
 }
 
-pub fn client_connecting(transport: Option<Res<NetcodeClientTransport>>) -> bool {
-    match transport {
+pub fn client_connecting() -> impl FnMut(Option<Res<NetcodeClientTransport>>) -> bool {
+    |transport| match transport {
         Some(transport) => transport.is_connecting(),
         None => false,
     }
 }
+
+pub fn client_just_connected() -> impl FnMut(Local<bool>, Option<Res<NetcodeClientTransport>>) -> bool {
+    |mut last_connected: Local<bool>, transport| {
+        let connected = transport.filter(|transport| transport.is_connected()).is_some();
+
+        if !*last_connected && connected {
+            *last_connected = true;
+            true
+        } else {
+            *last_connected = connected;
+            false
+        }
```

Comment:
```suggestion
let Some(transport) = transport else {
   return false;
};

let connected = transport.is_connected();
let just_connected = !*last_connected && connected;
*last_connected = connected;
just_connected
```

Same suggestion applies for the disconnected

Reply:
I like it this way a lot, thanks!